### PR TITLE
Add support for manipulating user status

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/main/java/com/nextcloud/operations/DeleteMethod.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* Nextcloud Android Library is available under MIT license
+ *
+ *   @author Tobias Kaminsky
+ *   Copyright (C) 2019 Tobias Kaminsky
```
Comment (AndyScherzinger) [inline src/main/java/com/nextcloud/operations/DeleteMethod.kt:4]:
```suggestion
 *   Copyright (C) 2020 Tobias Kaminsky
```

File: src/main/java/com/nextcloud/operations/PutMethod.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/* Nextcloud Android Library is available under MIT license
+ *
+ *   @author Tobias Kaminsky
+ *   Copyright (C) 2019 Tobias Kaminsky
```
Comment (AndyScherzinger) [inline src/main/java/com/nextcloud/operations/PutMethod.kt:4]:
```suggestion
 *   Copyright (C) 2020 Tobias Kaminsky
```

File: src/main/java/com/owncloud/android/lib/resources/users/GetPredefinedStatusesRemoteOperation.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+import org.apache.commons.httpclient.HttpStatus;
+
+import java.util.ArrayList;
+
+
+/**
+ * Remote operation performing the fetch of the public key for an user
+ */
+
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/GetPredefinedStatusesRemoteOperation.java:46]: remove empty line

File: src/main/java/com/owncloud/android/lib/resources/users/GetStatusRemoteOperation.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import com.owncloud.android.lib.ocs.ServerResponse;
+import com.owncloud.android.lib.resources.OCSRemoteOperation;
+
+import org.apache.commons.httpclient.HttpStatus;
+
+
+/**
+ * Remote operation performing the fetch of the public key for an user
+ */
+
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/GetStatusRemoteOperation.java:44]: remove empty line

File: src/main/java/com/owncloud/android/lib/resources/users/SetPredefinedCustomStatusMessageRemoteOperation.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.commons.httpclient.HttpStatus;
+
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+
+
+/**
+ * Remote operation performing the fetch of the public key for an user
+ */
+
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/SetPredefinedCustomStatusMessageRemoteOperation.java:45]: remove empty line

File: src/main/java/com/owncloud/android/lib/resources/users/SetStatusRemoteOperation.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.commons.httpclient.HttpStatus;
+
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+
+
+/**
+ * Remote operation performing the fetch of the public key for an user
+ */
+
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/SetStatusRemoteOperation.java:45]: remove empty line

File: src/main/java/com/owncloud/android/lib/resources/users/SetStatusRemoteOperation.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/SetStatusRemoteOperation.java:25]: remove empty line

File: src/main/java/com/owncloud/android/lib/resources/users/SetUserDefinedCustomStatusMessageRemoteOperation.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
+/* Nextcloud Android Library is available under MIT license
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/SetUserDefinedCustomStatusMessageRemoteOperation.java:1]: add line break

File: src/main/java/com/owncloud/android/lib/resources/users/SetUserDefinedCustomStatusMessageRemoteOperation.java [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+import org.apache.commons.httpclient.HttpStatus;
+
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+
+
+/**
+ * Remote operation performing the fetch of the public key for an user
+ */
+
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/SetUserDefinedCustomStatusMessageRemoteOperation.java:45]: remove empty line

File: src/main/java/com/owncloud/android/lib/resources/users/StatusType.kt [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+ *  
+ */
+
+package com.owncloud.android.lib.resources.users
+
+import com.google.gson.annotations.SerializedName
+
+enum class StatusType {
+    @SerializedName("online")
+    snline,
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/StatusType.kt:29]:
```suggestion
    online,
```

Typo?

Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/lib/resources/users/StatusType.kt:29]: Generally speaking: kt enum values should start with an upper case letter

**Review by AndyScherzinger — APPROVED** [review]

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger thanks for review, I think, I have updated everything

File: src/main/java/com/owncloud/android/lib/resources/users/StatusType.kt [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+ *  
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *  
+ */
+
+package com.owncloud.android.lib.resources.users
+
+enum class StatusType {
+    online,
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/lib/resources/users/StatusType.kt:26]: Afaik they should be starting with an upper case letter, no?

**Review by AndyScherzinger — APPROVED** [review]

**Comment by AndyScherzinger** [comment]
JUnit keeps failing

**Comment by tobiasKaminsky** [comment]
> JUnit keeps failing

Fixed. If no status was set ever before, endpoint returns 404 instead of empty status… :roll_eyes:

**Review by AndyScherzinger — APPROVED** [review]

**Pull request merged**